demisto_sdk/commands/test_content/tests/test_context_test.py

```diff
@@ -44,49 +44,297 @@ def test_is_runnable_on_this_instance(mocker):
                                              content_conf_json=content_conf_json,
                                              filtered_tests_content=filtered_tests)
-    mocked_demisto_client = DemistoClientMock(integrations=['mocked_integration'])
+    mocked_demisto_client = DemistoClientMock(integrations=[integrations_type])
     server_context = generate_mocked_server_context(build_context, mocked_demisto_client, mocker)
+    mocker.patch.object(server_context, mock_func, return_value=None)
+
+    return build_context, server_context
+
+
+def test_unmockable_playbook_passes_on_first_run(mocker, tmp_path):
+    """
+    Given:
+        - An unmockable test
+    When:
+        - The unmockable test passes on the first run
+    Then:
+        - Ensure that it exists in the succeeded_playbooks set
+        - Ensure that it does not exist in the failed_playbook set
+    """
+    execution_results = [PB_Status.COMPLETED]
+    build_context, server_context = init_server_context(mocker, tmp_path)
+    incident_test_mock = RunIncidentTestMock(execution_results)
     mocker.patch('demisto_sdk.commands.test_content.TestContentClasses.TestContext._run_incident_test',
-                 RunIncidentTestMock.run_incident_test)
+                 incident_test_mock.run_incident_test)
     server_context.execute_tests()
-    assert 'mocked_playbook (Second Playback)' in build_context.tests_data_keeper.failed_playbooks
-    assert 'mocked_playbook' not in build_context.tests_data_keeper.succeeded_playbooks
+    assert incident_test_mock.call_count == 1
+    assert not build_context.tests_data_keeper.failed_playbooks  # empty set
+    assert 'unmocked_playbook - Successful runs: 1/1' in build_context.tests_data_keeper.succeeded_playbooks
+
+
+def test_unmockable_playbook_passes_most_of_the_time(mocker, tmp_path):
+    """
+    Given:
+        - An unmockable test
+    When:
+        - The unmockable test passes on the second and third run
+    Then:
+        - Ensure that it exists in the succeeded_playbooks set
+        - Ensure that it does not exist in the failed_playbook set
+    """
+    execution_results = [PB_Status.FAILED, PB_Status.COMPLETED, PB_Status.COMPLETED]
+    build_context, server_context = init_server_context(mocker, tmp_path)
+    incident_test_mock = RunIncidentTestMock(execution_results)
+    logs = build_context.logging_module = mocker.MagicMock()
+    mocker.patch('demisto_sdk.commands.test_content.TestContentClasses.TestContext._run_incident_test',
+                 incident_test_mock.run_incident_test)
+    server_context.execute_tests()
+
+    assert incident_test_mock.call_count == 3
+    assert not build_context.tests_data_keeper.failed_playbooks
+    assert 'unmocked_playbook - Successful runs: 2/3' in build_context.tests_data_keeper.succeeded_playbooks
+    assert 'Test-Playbook was executed 3 times, and passed 2 times. Adding to succeeded playbooks.' == logs.info.call_args_list[14][0][0]
```

Comment:
Check if any of the calls contain the log message instead of a specific one.

Reply:
Done